pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -856,6 +856,183 @@ class ClassTypeHintMatch:
             ],
         ),
     ],
+    "climate": [
+        ClassTypeHintMatch(
+            base_class="Entity",
+            matches=_ENTITY_MATCH,
+        ),
+        ClassTypeHintMatch(
+            base_class="ClimateEntity",
+            matches=[
+                TypeHintMatch(
+                    function_name="state",
+                    return_type=["str", None],
+                ),
```

Comment:
State is declared final and should not occur.

Reply:
```suggestion
```